### PR TITLE
CakePHP 4 support

### PR DESCRIPTION
**Pull request by robertpustulka**
CakePHP 4+ compatibility updates.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,13 +12,13 @@
         }
     ],
     "require": {
-        "cakephp/cakephp": "^3.5",
+        "cakephp/cakephp": "4.*",
         "php": ">=5.6"
```
Comment (chris48s) [inline composer.json:16]: I think this should now be
`"php": ">=7.2"`

**Comment by chris48s** [comment]
Hi @robertpustulka

Thanks for having a look at this.

I wrote a few CakePHP plugins a few years back, but since 2017 I don't use PHP in my job at all any more. This means I have not really kept pace with the latest developments in the PHP or CakePHP communities. I've had a look and the changes in this PR look sensible to me and all the tests pass, but I've never used CakePHP4 at all myself so I suspect if there was a small error I would not see it.

So here is my suggestion.. I notice you have made substantial contributions in the CakePHP ecosystem including to CakePHP core. If you are still relying on this library, I would be happy to either:

* Give you push access to this repo and maintainer access to the package on packagist
or
* You can make a fork, push a new package to packagist and I'll mark https://packagist.org/packages/chris48s/cakephp-searchable abandoned in favour of your new package

This is a small module which not require much attention (I made the repo in 2016 and this is issue #4 🙂) and if you are actively using it you are better placed to keep it up to date than I am.

What do you think?

**Comment by umer936** [comment]
Hi. 

I've tried adding the plugin from @robertpustulka 's "cake-4" branch in my composer.json using the repository section and it seems to include, but I can't get it to find the Behaviors... 

However, if either of the things @chris48s said happen, then I can remove all that and just include the package normally, which would be nice :)

**Comment by umer936** [comment]
Here's composer.json to be able to use it: 
```
    "repositories": [
        {
            "type": "package",
            "package": {
                "name": "chris48s/cakephp-searchable",
                "version": "2.0",
                "type": "package",
                "source": {
                    "url": "https://github.com/robertpustulka/cakephp-searchable.git",
                    "type": "git",
                    "reference": "cake-4"
                }
            }
        }
    ],
```
```
    "require": {
        "chris48s/cakephp-searchable": "^2.0"
    },
```
```
    "autoload": {
        "psr-4": {
            "Chris48s\\Searchable\\": "./vendor/chris48s/cakephp-searchable/src/"
        }
    },
```
```
    "extra":
    {
        "installer-paths":
        {
            "vendor/chris48s/cakephp-searchable": ["Chris48s/Searchable"]
        }
    },
```
and bootstrap()
```
        $this->addPlugin('Chris48s/Searchable',  [
            'path' => ROOT . DS . 'vendor' . DS . 'chris48s' . DS . 'cakephp-searchable' . DS . 'src'
        ]);
```

Not sure if all of these bits are correct, but it does finally work with all of these sections

**Comment by robertpustulka** [comment]
@chris48s 
Hi! Sorry for the late response. You can give me push access as a start, we'll see how it goes from here :). Thanks!

@umer936 
Adding this to repositories should be enough:
```json
        {
            "type": "vcs",
            "url": "https://github.com/robertpustulka/cakephp-searchable.git"
        }
```

My fix is tagged as 3.0:

```
"chris48s/cakephp-searchable": "^3.0"
```

CakePHP plugin installer will figure out the path itself. Just load a plugin as usual.
```php
$this->addPlugin('Chris48s/Searchable');
```

**Comment by chris48s** [comment]
No problem :+1:

- I've invited you to this repo. Once you accept the invite I'll make sure you have access to push and tag
- The package is set to auto-update from github, so pushing a tag to the repo should also create a release on packagist but let me know your packagist username if there is an issue. I can add you as a maintainer on the package as well we need to
- The next release should be 3.0.0 as dropping CakePHP 3 is a non-backwards-compatible change. I've pushed a commit to this branch starting a changelog entry: 405eb54
- Feel free to merge and tag when you're ready

Thanks

**Comment by wate** [comment]
@robertpustulka 

Would you like to be merged?

**Comment by robertpustulka** [comment]
@wate merged

I no longer use this package sorry. But the fix introduced here should work, at least it worked while I was using it.

**Pull request merged**